xdsl/ir/core.py

```diff
@@ -1118,6 +1118,7 @@ class BlockCallback(Protocol):
         def __call__(self, *args: BlockArgument) -> list[Operation]:
             ...
 
+    @deprecated("Please use Builder instead")
```

Comment:
I'm not sure we want to deprecate this now?
At least, I don't see any replacement for it, as the only way is to either do it by hand or create an implicit builder.
Can we instead make `BlockCallback` take a builder as argument or something?

Reply:
But there is a replacement for it, which is to create it by hand or use an implicit builder :)

Reply:
I'd like to delete BlockCallback as soon as clients have migrated off

Reply:
Well, I think that's a problem to have only these two options. I feel that in general, we should have 
* The hand-way, which realistically no one should use
* The manual builder way, which is the MLIR-like way
* The implicit builder way, which is the nice xDSL way

But I'm actually fine with this PR, since none of these cases should require the `from_callable`